src/fluent-plugin-mdsd/lib/fluent/plugin/out_mdsd.rb

```diff
@@ -62,7 +64,11 @@ def shutdown()
         # NOTE: a plugin must define this because base class doesn't have
         # default implementation.
         def format(tag, time, record)
-            [tag, time, record].to_msgpack
+            if @use_source_timestamp
+                [tag, time, record].to_msgpack
+            else
+                [tag, Fluent::Engine.now, record].to_msgpack
```

Comment:
I'm not a 100% sure this should be done here or if it should be done in "write"

Reply:
the 'time' is an integer (unix time_t). I am not sure what's "Fluent::Engine::now", but  Time.now.tv_sec should work


---
In reply to: [136669547](https://github.com/Azure/fluentd-plugin-mdsd/pull/39#discussion_r136669547) [](ancestors = 136669547)

Reply:
https://github.com/fluent/fluentd/blob/master/lib/fluent/engine.rb#L190

I could change it to Time.now.to_i or Time.now.tv_sec if you prefer